### ISSUE TITLE
Implement `update` functionality for `RouteTable` resource

### ISSUE THREAD

**Issue by brycahta** [aws-controllers-k8s/community#1201]
**Is your feature request related to a problem?**
[ec2-controller](https://github.com/aws-controllers-k8s/ec2-controller) supports Create/Delete functionality for `RouteTable` resource, but **not** [Update](https://github.com/aws-controllers-k8s/ec2-controller/blob/af432344f155be0761ed68724ff04c04b6e89bcf/pkg/resource/route_table/sdk.go#L462). _Note: although a customUpdateRouteTable() method is defined here, this was necessary to add Routes to the RouteTable Spec NOT update RouteTable itself_

**Describe the solution you'd like**
I should be able to modify/update an ACK-managed `RouteTable` resource using its controller

**Describe alternatives you've considered**
Updating resource myself using `aws-sdk`, cli, or console

**Comment by brycahta** [comment]
![image](https://user-images.githubusercontent.com/61433408/188704195-fd079f8e-2515-4f4a-967e-51df8b117174.png)


* Only [`Edit routes`](https://github.com/aws-controllers-k8s/ec2-controller/blob/22e80dc1bf371c0be9abff8727bf68f889d88f86/pkg/resource/route_table/hooks.go#L198) from the list of _Console Actions_ is implemented

* This issue is **blocking**: https://github.com/aws-controllers-k8s/community/issues/1291
  * TLDR; `RouteTable.Spec.Routes` is not updated properly when using _references_
